model/impl.go

```diff
@@ -80,6 +85,21 @@ func (rc *realModel) addNode(hostname string) *NodeInfo {
 	return node_ptr
 }
 
+// Deletes the given node with all of the pods and containers on it.
+func (rc *realModel) deleteNode(hostname string) {
+	rc.lock.Lock()
+	defer rc.lock.Unlock()
+
+	if node, ok := rc.Nodes[hostname]; ok {
+		for _, podInfo := range node.Pods {
+			if namespaceInfo, ok := rc.Namespaces[podInfo.Namespace]; ok {
```

Comment:
That's fine. The problem may be solved by delegating the responsibilty of locking to the caller, but maybe this is won't simplify code here.